web/src/handler/types/query.rs

```diff
@@ -0,0 +1,48 @@
+use std::{convert::Infallible, future::Future};
+
+use serde::Deserialize;
+
+use crate::{handler::FromRequest, request::WebRequest};
+
+pub struct Query<T>(pub T);
+
+impl<'a, 'r, S, T> FromRequest<'a, WebRequest<'r, S>> for Query<T>
+where
+    S: 'r,
+    T: for<'de> Deserialize<'de>,
```

Comment:
This line effectively implies `T: 'static`, because it requires `T` to be created from any possible lifetime `'de` of the request. Indeed it's the very definition of `DeserializeOwned` if you look it up.

But I think I have an idea..